src/container/container.ts

```diff
@@ -106,7 +110,8 @@ class Container implements interfaces.Container {
         this._deactivations = new Lookup<interfaces.BindingDeactivation<any>>();
         this.parent = null;
         this._metadataReader = new MetadataReader();
-        this._moduleActivationStore = new ModuleActivationStore()
+        this._moduleActivationStore = new ModuleActivationStore();
+        this._contextStack = new ContextStack(this)
```

Comment:
The ContextStack provides the context hierarchy in conjunction with planAndResolve.
What I have not done is deal with snapshot and restore.  The ContextStack uses the Container.options to stay current.
We currently do not snapshot and restore container options which seems incorerct to me.

Thoughts ?

Reply:
It also has not been tested as a unit.